### PR TITLE
feat: add new threat shield configurations

### PR DESCRIPTION
**Pull request by andre8244**
Ref:
- https://github.com/NethServer/nethsecurity/issues/795

### CODE REVIEW

**Comment by andre8244** [comment]
Let's fix *Lorem ipsum*s before merging

File: public/i18n/en/translation.json [hunk 1]
```diff
@@ -1454,17 +1459,53 @@
...
+      "blocked_addresses": "Blocked addresses",
+      "allowed_addresses": "Allowed addresses",
+      "logging": "Logging",
+      "logging_description": "Threats monitoring will take account and show only threats that have been logged.",
+      "log_prerouting_chain": "Log packets blocked in pre-routing chain",
+      "log_input_chain": "Log packets blocked in input chain",
+      "log_forward_chain": "Log packets blocked in forward chain",
+      "log_forward_lan_chain": "Log packets blocked forwarded from LAN",
+      "block_brute_force": "Block brute force attacks",
+      "block_brute_force_description": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1485]: Brute force attack protection safeguards against repeated unauthorized login attempts by blocking or limiting access after multiple failed attempts.

File: public/i18n/en/translation.json [hunk 2]
```diff
@@ -1454,17 +1459,53 @@
...
+      "block_brute_force_tooltip": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1486]:
```suggestion
      "block_brute_force_tooltip": "When enabled, the system will scan logs for patterns of failed access attempts.",
```

File: public/i18n/en/translation.json [hunk 3]
```diff
@@ -1454,17 +1459,53 @@
...
+      "ban_after_n_attempts": "Ban after N failed accesses",
+      "ban_after_n_attempts_tooltip": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1488]:
```suggestion
      "ban_after_n_attempts_tooltip": "The number of times an IP must appear in the log before being flagged as suspicious and blocked.",
```

File: public/i18n/en/translation.json [hunk 4]
```diff
@@ -1454,17 +1459,53 @@
...
+      "attack_patterns": "Patterns to detect attacks",
+      "attack_patterns_tooltip": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1490]:
```suggestion
      "attack_patterns_tooltip": "Enter a regular expression to search the logs; if matched, the source IP will be marked as suspicious.",
```

File: public/i18n/en/translation.json [hunk 5]
```diff
@@ -1454,17 +1459,53 @@
...
+      "add_pattern": "Add pattern",
+      "block_dos": "Block DoS",
+      "block_dos_description": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1493]:
```suggestion
      "block_dos_description": "DoS (Denial of Serevice) limits excessive suspicious requests from a single source",
```

File: public/i18n/en/translation.json [hunk 6]
```diff
@@ -1454,17 +1459,53 @@
...
+      "block_icmp_dos": "Block ICMP DoS",
+      "block_syn_dos": "Block SYN DoS",
+      "block_udp_dos": "Block UDP DoS",
+      "threats_monitoring_disabled": "Threats monitoring disabled",
+      "threats_monitoring_disabled_message": "Threats monitoring shows only logged threats. Enable logging on at least one chain to enable threats monitoring.",
+      "enter_a_number_greater_than_0": "Enter a number greater than 0",
+      "ban_time": "Ban time",
+      "ban_time_tooltip": "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed do eiusmod tempor incidunt ut labore et dolore magna aliqua.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1501]:
```suggestion
      "ban_time_tooltip": "After the timeout period, the blocked IP will be automatically unblocked, restoring its access to the system.",
```

File: src/components/standalone/dashboard/ThreatShieldIpCard.vue [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+            kind="warning"
+            :text="t('standalone.dashboard.monitoring_disabled')"
+            :icon="faWarning"
+          />
+        </template>
+        <template #content>
+          <i18n-t keypath="standalone.dashboard.monitoring_disabled_tooltip" tag="p" scope="global">
+            <template #page>
+              <NeLink invertedTheme @click="goTo('/security/threat-shield?tab=settings')">
+                {{ t('standalone.threat_shield.settings') }}
```
Comment (Tbaile) [inline src/components/standalone/dashboard/ThreatShieldIpCard.vue:204]: I'd use a RouterLink component here (naturally wrapped by a NeLink for styling)
https://router.vuejs.org/guide/#App-vue

File: src/views/standalone/security/ThreatShieldView.vue [hunk 8]
```diff
@@ -10,6 +10,12 @@ import { useTabs } from '@/composables/useTabs'
 import AllowlistTab from '@/components/standalone/security/threat_shield/AllowlistTab.vue'
 import BlocklistTab from '@/components/standalone/security/threat_shield/BlocklistTab.vue'
 import SettingsTab from '@/components/standalone/security/threat_shield/SettingsTab.vue'
+import LocalBlocklistTab from '@/components/standalone/security/threat_shield/LocalBlocklistTab.vue'
+
+export type BlockOrAllowAddress = {
+  address: string
+  description: string
+}
```
Comment (Tbaile) [inline src/views/standalone/security/ThreatShieldView.vue:18]: Maybe `BanIpLocalAddress` is better?

It's your call on this one

Comment (andre8244) [reply src/views/standalone/security/ThreatShieldView.vue]: Done

**Review by Tbaile — APPROVED** [review]

**Pull request merged**